### ISSUE TITLE
How could I use glide to store dependencies in the different directory?

### ISSUE THREAD

**Issue by gobwas** [Masterminds/glide#452]
Hi! 

I want to store my codebase with vendored dependencies *not* inside the `GOPATH` directory.
So I need to create tree like this:
```
├── src
│   └── my_app
│       └── my_app.go
├── vendor
│   └── src
│       └── my_dep/...
```

And then, to use glide for serving dependencies into the `vendor/src`, I want put `glide.yaml` right there:

```
├── src/...
├── vendor/...
├── glide.yaml
```

Is it possible to use glide in this way?

**Comment by mattfarina** [comment]
`go` is what resolves the package locations. Glide is simply a helper that puts packages where Go expects them and helps you manage things like the versions of the packages. You can't store your code in at structure because `go` won't resolve it there. This has nothing to do with Glide and is about Go resolution.

**Comment by sdboyer** [comment]
unless i'm missing something (and it's late, so that's not unlikely), it seems like the only problem with @gobwas's requested structure is that he wants to put deps under `vendor/src`, rather than under `vendor` itself.

That doesn't seem like the substance of the request, though. It seems like the main ask here is to have `glide.yaml` in a parent dir of his own project's go code, and have that control the subdirs. That, I think, should work fine.

**Comment by gobwas** [comment]
@sdboyer yes, you are completely right =)
Some `vendor_dir` option in `yaml` configuration for glide would be very useful for me =)

**Comment by gobwas** [comment]
@mattfarina you could simply say `GOPATH=$PWD/vendor:$PWD go build` to simply switch resolving paths for go. And thats why feature for changing vendored packages dir in `glide` also be good =) I could prepare PR if this makes sense.

**Comment by sdboyer** [comment]
@gobwas no, I think we're back to a misunderstanding now. `glide` does not mess with `GOPATH`, and glide does not allow a configurable `vendor/` directory, because as @mattfarina said, that IS a product of how Go works. We can't, and don't want to, change that.

**Comment by sdboyer** [comment]
mmm, i see how my comment left ambiguity, though:

> it seems like the only problem with @gobwas's requested structure is that he wants to put deps under `vendor/src`, rather than under `vendor` itself.

> That doesn't seem like the substance of the request, though...

What I was saying here is that that's simply not possible, as it's not how vendor works - the specification entails that there is no `src` dir under `vendor`, unlike with normal `GOPATH` where everything lives at `GOPATH/src/<import path>`. And glide doesn't do `GOPATH` manipulation, because other tools over the years have made it amply clear how brittle that makes your environment.

What I was trying to say, @gobwas, is that as long as you drop that `src/` dir from under `vendor`, then then the corresponding `glide.yaml` at the root will cover deps for everything under `src/my_app`.
